lib/instrumentation/index.js

```diff
@@ -94,21 +94,28 @@ Instrumentation.prototype.addPatch = function (modules, handler) {
     }
 
     this._patches.add(mod, handler)
-    this._startHook()
```

Comment:
I don't think there was any particular reason why this was done inside of the loop, so I moved it outside.

Reply:
That's fine. It only needs to be triggered before returning to user code, so it can be just once at the end. 👍